### PR TITLE
Showing step line no in JSON logger #getgauge/intellij-plugin120

### PR DESCRIPTION
**Pull request by BugDiver**
listening StepEnd even and caching steps corresponding to current
scenario.

### CODE REVIEW

**Comment by gaugeci** [comment]
@cla-bot check if all the contributors of this pull request has signed our CLA.

File: reporter/jsonConsole.go [hunk 1]
```diff
@@ -84,9 +89,12 @@ type executionError struct {
 	Text       string `json:"text"`
 	Filename   string `json:"filename"`
 	Message    string `json:"message"`
+	LineNo     string `json:"lineNo"`
 	StackTrace string `json:"stackTrace"`
 }
 
+var stepCache = make(map[*gm.ScenarioInfo][]*stepInfo)
```
Comment (kashishm) [inline reporter/jsonConsole.go]: This cache can be added to JSONConsole instead of a global var.

Comment (BugDiver) [reply reporter/jsonConsole.go]: In that case, the private methods who are using the cache has to be bound with JSONConsole

Comment (kashishm) [reply reporter/jsonConsole.go]: Or it can be passed as param.

File: reporter/jsonConsole.go [hunk 2]
```diff
@@ -200,6 +207,10 @@ func (c *jsonConsole) StepStart(stepText string) {
 }
 
 func (c *jsonConsole) StepEnd(step gauge.Step, res result.Result, execInfo gm.ExecutionInfo) {
+	si := new(stepInfo)
```
Comment (kashishm) [inline reporter/jsonConsole.go]: Simplify this to `&stepInfo{step: &step, protoStep: res.(*result.StepResult).Item().(*gm.ProtoStep)}`

File: reporter/jsonConsole.go [hunk 3]
```diff
@@ -183,7 +190,7 @@ func (c *jsonConsole) ScenarioEnd(scenario *gauge.Scenario, res result.Result, i
 		Res: &executionResult{
 			Status:            getScenarioStatus(res.(*result.ScenarioResult)),
 			Time:              res.ExecTime(),
-			Errors:            getErrors(getAllStepsFromScenario(res.(*result.ScenarioResult).ProtoScenario), i.CurrentSpec.FileName),
+			Errors:            getErrors(c, getAllStepsFromScenario(res.(*result.ScenarioResult).ProtoScenario), i.CurrentSpec.FileName, i),
```
Comment (kashishm) [inline reporter/jsonConsole.go]: Is there any specific reason to pass the JSONConsole instead of passing the cache?

Comment (BugDiver) [reply reporter/jsonConsole.go]: No. We can pass `stepCache` instead of `JSONCosole`.

**Pull request merged**